introducing-signal-store.md

```diff
@@ -0,0 +1,701 @@
+Welcome to **MiniRx Signal Store**, the new state management library from [MiniRx](https://mini-rx.io/).
+
+With the arrival of [Signals](https://angular.io/guide/signals) in Angular 16, new best practices are arising for state and events. In the future Signals and Observables will coexist in your application.
+
+Modern Angular needs modern state management which promotes **new Angular best practices** and **streamlines the usage of Signals and Observables**...
+
+MiniRx Signal Store is exactly that.
+
+### Modern state management with MiniRx Signal Store:
+
+* Signal Store is an **[Angular](https://angular.dev/)-only** state management library
+* Signal Store **embraces [Angular Signals](https://angular.io/guide/signals)** and leverages **Modern Angular APIs** internally
+* Signal Store is based on the same great concept as the original **[MiniRx Store](https://mini-rx.io/)**
+  * Manage **global** state at large scale with the **Store (Redux) API**
+  * Manage **global** state with a minimum of boilerplate using **Feature Stores**
+  * Manage **local** component state with **Component Stores**
+  * All combined in a single library
+  * MiniRx always tries to find the sweet spot between powerful, simple and [lightweight](https://github.com/spierala/angular-state-management-comparison)
+* Signal Store implements and promotes new **Angular best practices**:
+  * **Signals** are used for **(synchronous) state**
+  * **RxJS** is used for events and **asynchronous tasks**
+* Immutable Signal State: Immutability can be enforced with the Immutable State extension
+* Signal Store helps to streamline your usage of [RxJS](https://rxjs.dev/) and [Signals](https://angular.io/guide/signals): e.g. `connect` and `rxEffect` understand both Signals and Observables
+* Signal Store has first-class support for OOP style (e.g. `MyStore extends FeatureStore`), but offers also functional creation methods (e.g. `createFeatureStore`)
+* Simple refactor: If you used MiniRx Store before, refactor to Signal Store will be straight-forward: change the TypeScript imports, remove the Angular async pipes (and ugly non-null assertions (`!`)) from the template
+
+### Getting Started
+To install the @mini-rx/signal-store package, use your package manager of choice:
+
+`npm install @mini-rx/signal-store`
+
+### Use-cases
+
+MiniRx Signal Store is highly flexible and offers three different well-defined state containers out of the box:
+
+- Store (Redux)
+- Feature Store
+- Component Store
+
+All three can be easily used together in your application.
+Depending on the use-case, you can choose the state container which suits your needs.
+
+These are the typical use-cases:
+
+![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/qj5anf3ibwrjw20yu6t0.png)
```

Comment:
Do you have your own blog? A successful community leader told me (as I was posting on dev.to as well) that it's not good in the long run, as it gives me/you nothing.

Reply:
No own blog yet and no plans to do so...

Currently, I am happy with dev.to
But it is crucial that we can post e.g. under This Is Angular.
Otherwise your posts would just disappear in the mass of posts on dev.to

I think that I can be a challenge as well, to build a successful own blog which finds its audience.

Reply:
@spierala start small - here's a few-hours what you can do:
- create a repo on github pages + configure it - few minutes ;)
- use this free developer blog template - https://dev.to/danielcgilibert/free-open-source-blog-template-for-developers-218e
- publish spierala.github.io
- make sure `canonical` is set: `<link rel="canonical" href="https://ducin.dev/typescript-type-inference-guide">`
Then:
- first - publish on your blog post
- 1 day later - REpublish on dev.to and set canonical to your website there

this way you slowly grow your own blog for a price of few hours + you DON'T lose dev.to traffic.

Reply:
spierala.github.io sounds good for a start ;)

<img width="1363" alt="image" src="https://github.com/spierala/blog-posts/assets/375027/07df5699-0616-4963-ace2-b89cf88964d6">

spierala.dev (like ducin.dev I bought last week) - costs 10 USD/year

Reply:
dev.to - canonical:

<img width="862" alt="image" src="https://github.com/spierala/blog-posts/assets/375027/fe5d31b3-42db-4035-9906-47a3d3e8e24d">

Reply:
I would very much vote against picking your own blog over dev.to if you do not want to spend time and effort into building an audience for it. However, it's a good suggestion to use the canonical URL if you _do_ want to make that investment. But if you don't, users will end up on your site but they won't have any incentive to stay and if it doesn't look highly professional, it's just gonna come across as amateuristic. In that case, I think it's better to let it sit on dev.to, where there's already a lot of functionality built-in (comments, moderation, tags ...) and there's other content (potentially other content from you as well) that could interest readers.

Reply:
own blog is for later.

For now dev.to does the job :)

But thanks for the input, I did not think of an own blog so far.